src/main/java/isaac/bastion/BastionGroup.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Created by Aleksey on 11.07.2017.
+ */
+
+package isaac.bastion;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class BastionGroup {
+	private int groupId;
+	private Set<Integer> allowedGroupIds;
```

Comment:
Support for this is already handled, for example:

https://github.com/DevotedMC/Bastion/pull/38/files#diff-b530412c860d5e17c39b3300041bbcb5R134

https://github.com/DevotedMC/Bastion/pull/38/files#diff-b530412c860d5e17c39b3300041bbcb5R280

But I will do more testing together when I will test my changes in CastleGates to support "allowed groups"
I will test CG when current changes will be accepted and "interface" approved